content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+
+    expect(response_body).to eq([
+      { "title" => "Test Driven Development" },
+      { "title" => "Patterns of Enterprise Application Architecture" }
+    ])
+  end
+end
+```
+
+If you run this test, you'll see that it fails because our application currently returns a 404 response for the `/books` route.
+
+Let's fix that by generating an action for a books index:
+
+```shell
+bundle exec hanami generate action books.index
+```
+
+In addition to generating an action at `app/actions/books/index.rb`, the generator has also added a route in `config/routes.rb`:
+
+```ruby
+# frozen_string_literal: true
+
+module Bookshelf
+  class Routes < Hanami::Routes
+    root to: "home.index"
+    get "/books", to: "books.index"
+  end
+end
+```
+
+If we run our spec again, our expectation for a successfull response is now satisfied, but there's a different failure:
+
+```shell
+bundle exec rspec spec/requests/books/index_spec.rb
+
+Randomized with seed 58022
+
+GET /books
+  returns a list of books (FAILED - 1)
+
+Failures:
+
+  1) GET /books returns a list of books
+     Failure/Error: expect(last_response.content_type).to eq("application/json")
+
+       expected: "application/json"
+            got: "text/html; charset=utf-8"
+
+       (compared using ==)
+     # ./spec/requests/books/index_spec.rb:8:in `block (2 levels) in <top (required)>'
+```
+
+Our response doesn't have the expected format. Let's adjust our action to return a JSON formatted response using `response.format = :json`. We'll also set the response body to what our test expects:
+
+```ruby
+# app/actions/books/index.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        def handle(*, response)
+          books = [
+            {title: "Test Driven Development"},
+            {title: "Patterns of Enterprise Application Architecture"}
+          ]
+
+          response.format = :json
+          response.body = books.to_json
+        end
+      end
+    end
+  end
+end
+```
+
+If we run our spec, it now passes!
+
+```shell
+bundle exec rspec spec/requests/books/index_spec.rb
+
+Randomized with seed 16566
+
+GET /books
+  returns a list of books
+
+Finished in 0.02378 seconds (files took 0.49411 seconds to load)
+1 example, 0 failures
+```
+
+## Persisting books
+
+Of course, returning a static list of books is not particularly useful.
+
+Let's address this by retrieving books from a database.
+
+<p class="notice">
+  Integrated support for peristence based on <a href="https://rom-rb.org/">rom-rb</a> is coming in Hanami's 2.1 release. For now, we can bring our own simple rom-rb configuration to allow us to store books in a database.
+</p>
+
+### Adding persistence using rom-rb
+
+Let's add just enough rom-rb to get persistence working using Postgres.
+
+First, add these gems to the Gemfile and run `bundle install`:
+
+```ruby
+# Gemfile
+gem "rom"
+gem "rom-sql"
```

Comment:
The code in guide will work only with `3.6.0` (as of now) - does not work with 3.5.0 IIRC (or rather, it won't resolve to 3.5.0) and definitely won't work with 4.0. Perhaps it's worth to pin the version?

Reply:
Ah, right - thank you. I've gone with:

```
gem "rom", "~> 5.3"
gem "rom-sql", "~> 3.6"
```